### ISSUE TITLE
content stripped of elements when rendered

### ISSUE THREAD

**Issue by RobertChristopher** [devpunks/snuggsi#178]
hey, the template is rendering the content as plain text, stripping the <tr> / <td> elements. @snuggs 
 
```html
<script src=//unpkg.com/snuggsi></script>

<tbody> 
  <template name=proxies>
    <tr>
      <td>{ip}</td>
      <td>{port}</td>
      <td>{user}</td>
      <td>{pass}</td>
      <td>{alive}</td>
      <td>{speed}</td>
      <td>{website}</td>
    </tr>
  </template>
</tbody>

<script>

const
  template = Template `proxies`
, context  = [
    { 
      ip: '54.86.170.131',
      port: '33128',
      user: 'rob',
      pass: '1111',
      alive: true,
      speed: 111,
      website: 'https://sneakyhead.com'
    }, 

    {
      ip: '54.86.170.142',
      port: '33128',
      user: 'rob',
      pass: '1111',
      alive: true,
      speed: 222,
      website: 'https://supremenewyork.com'
    }
  ]

template.bind (context)

</script>
```

**Comment by RobertChristopher** [comment]
@snuggs very inspired by snuggsi & the philosophy here, its a large step away from the corporate code that powers the world wide web. I see myself as more of a `devpunk' haha & am inspired to deconstruct down to the platform layer. its a powerful foundation to reconstruct over & I've been shifting my perception of development. 

this gem by @tmornini  really woke me up. https://slack-files.com/T03JT4FC2-F151AAF7A-13fe6f98da

theres an art to building conventions and connecting them into larger flows. the medium of the web is shifting to web components & using the structure of links. a lot of these early web component frameworks are building complex solutions ( polymer ), which are early implementations. I'd prefer to plant my own seeds right on top of the platform & grow strong roots, building my own conventions. its an art of self expression thats not injected with corporate framework. 
 
frameworks are competitive but snuggsi is complementary which is why I've really enjoyed working with it. another project that has really opened my mind is cell.js ( https://www.celljs.org ). plain old javascript objects are simple. simple is better than complex!

@snuggs been trying to get in contact man & get a hangout. I'd love to start contributing to snuggsi as I plan to build an empire of components.

**Comment by tmornini** [comment]
@RobertChristopher Thanks for the mention, but I did write that post — though I agree with it — and I may have shared it, it feels familiar. 😎

**Comment by snuggs** [comment]
@RobertChristopher @tmornini found a good example of our scope. Because remember snuggsi is just automation of boilerplate to use what's already in the platform. Seems like we can iterate table rows so that's a good sign. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template

**Comment by RobertChristopher** [comment]
@snuggs exactly, very simple. i'll do some self digging around the codebase to become more familiar.

**Comment by RobertChristopher** [comment]
@snuggs intuition brought me into here https://github.com/devpunks/snuggsi/blob/master/token-list/index.es ...

**Comment by snuggs** [comment]
@RobertChristopher can you throw up a jsfiddle with a repro(duction) of the issue? Best part is you souldn't have to include any dependencies other than the HTML you provided.

@tmornini @brandondees @rianby64 and now we see the power of the library. #UseThePlatform

**Comment by snuggs** [comment]
@RobertChristopher I have been investigating this issue the past couple weeks and think we have a wrangle on things. We have some good news and bad news:

## Bad News
-  Internet Explorer will strip `<table>` specific child elements (i.e. `<tr>`,`<td>`, etc.) from any element not within a `<table>`, `<tbody>`, `<thead>`, nor `<tfoot>`. This **IS** a platform issue of **ALL** Internet Explorer versions _(excluding EDGE)_ P.S. ** [It’s Not Possible!](https://github.com/aurelia/html-template-element/issues/3) and seems like even [tests with lit-html were disabled for Internet Explorer](https://github.com/Polymer/lit-html/pull/212)
**

This is an active problem which creates need for [libraries to wrap html strings in appropriate parent tags](https://github.com/kevinokerlund/html-fragment).
[Internet Explorer 11 is now moved from Q1/2021 EOL to (sometime in)2021 :-()(https://github.com/SAP/openui5/issues/1697#issuecomment-448742165)
[The same issue](https://github.com/aurelia/binding/issues/595) exists for [`<select>` and `<option>`](https://stackoverflow.com/questions/51976172/polymer-ie11-dom-repeat-no-rendering-option-tags)


### Remedies
  - [ ] Create a detailed warning documentation when using `<template>`


## _(Now on to the)_ Good News


It is assumed a table cannot be created without a comment by using a template representing a partial aspect of the table.

Claims (Cannot create partial tables with templates):
github.com/WebReflection/hyperHTML/issues/194
github.com/WebReflection/hyperHTML/issues/79#issue-248628718

Claim: Partial attributes and attributes without quotes
https://github.com/WebReflection/hyperHTML/issues/79#issue-248628718

Our following examples hold this to not be true:

JSFiddle with table rows
JSFiddle with table cells
JSFiddle with partial attributes & attributes without comments


### Remedies
#### [ ] Use Ranges in `<template>`
Use a Range instead of multiple string objects for template iteration. This is not only cleaner but allows markers to be placed in code. This prevents the need to have comments be used in order to find placement of replaceable content within the element. This allows us to Use DocumentFragments in memory [which has been supported standard since jQuery](https://johnresig.com/blog/dom-documentfragments/). Best part is there is [support back to 13 November, 2000 since DOM Level 2](https://www.w3.org/TR/2000/REC-DOM-Level-2-Traversal-Range-20001113/ranges.html).

@RobertChristopher I have been investigating this issue the past couple weeks and think we have a wrangle on things. We have some good news and bad news:

## Bad News
-  Internet Explorer will strip `<table>` specific child elements (i.e. `<tr>`,`<td>`, etc.) from any element not within a `<table>`, `<tbody>`, `<thead>`, nor `<tfoot>`. This **IS** a platform issue of **ALL** Internet Explorer versions _(excluding EDGE)_ P.S. ** [It’s Not Possible!](https://github.com/aurelia/html-template-element/issues/3) and seems like even [tests with lit-html were disabled for Internet Explorer](https://github.com/Polymer/lit-html/pull/212)
**

### [ ] Use aria labeled custom elements for `<table>`s and `<select>`s
  References:
https://www.youtube.com/watch?v=7a6bLXw2lqQ
https://github.com/GoogleChromeLabs/howto-components

**Example**

```html

<!— Table —>
<aria-table role=table>
  <aria-thead>Header</aria-thead>
  <aria-tfoot>Foooter</aria-tfoot>

  <aria-tbody>
  <aria-tr role=row>
    <aria-td role=cell>Cells or rows can be repeated.</aria-td>
  </aria-tr>
  </aria-tbody>
</aria-table>

<!— Select —>
<aria-select role=listbox>
  <aria-optgroup role=listgroup>
    <aria-option role=listitem>Option Text</aria-option>
    <aria-option role=listitem>Option Text</aria-option>
  </aria-optgroup>

  <aria-option role=listitem selected>Option outside of group</aria-option>
</aria-select>
```

This will be a great choice as it is not only backwards compatible but forward thinking as well. Only question would be does this create an actual `<table>` and `<select>` for IE11?


#### Wrap `<template>` content in tables.
** WARNING: Avoid like the plague** _(for obvious reasons)_
https://stackoverflow.com/questions/26384954/template-tag-polyfill-for-ie-11-not-working-with-table-tr-and-td/31617318

#### Use `<tr>` as an ad-hock _ »template »_

#### Use a custom element. _(Why not just start using them now instead of the things that don’t work?)_
  See example: https://github.com/w3c/webcomponents/issues/765#issuecomment-419721136

Perhaps this may be a good use case - https://github.com/aurelia/html-template-element/issues/3#issuecomment-171041821
Great blog post on solution - http://davismj.me/blog/advanced-tables/

## References

  - [W3C DOM Level 2 Ranges](https://www.w3.org/TR/2000/REC-DOM-Level-2-Traversal-Range-20001113/ranges.html)
[webcomponents polyfill issue](https://github.com/webcomponents/template/issues/3)
[lit-html IE issue](https://github.com/Polymer/lit-html/issues/210)
[Polymer issue with IE](https://github.com/Polymer/polymer/issues/1567)
[Polymer IE 11 Issue with `dom-repeat`](https://github.com/Polymer/polymer/issues/1735)
[Juicy Templates Issue](https://github.com/Juicy/imported-template/issues/10)
